features/admin.sms-templates.v1/pages/sms-customization.tsx

```diff
@@ -0,0 +1,457 @@
+        }));
+    }, [ smsTemplateError ]);
+
+    // This is used to check whether the URL contains a template ID, and if so, set it as the selected template.
+    useEffect(() => {
+        const hash: string = window.location.hash;
+
+        if (hash.startsWith("#templateId=")) {
+            const templateId: string = hash.split("=")[1];
+
+            setSelectedSmsTemplateId(templateId);
+            setSelectedSmsTemplateDescription(availableSmsTemplatesList?.find(
+                (template: SmsTemplateType) => template.id === templateId)?.description);
+        }
+    }, [ window.location.hash ]);
+
+    const handleTemplateIdChange = (templateId: string) => {
+        setIsTemplateNotAvailable(false);
+        setCurrentSmsTemplate(undefined);
+        setSelectedLocale(I18nConstants.DEFAULT_FALLBACK_LANGUAGE);
+        setSelectedSmsTemplateId(templateId);
+        setSelectedSmsTemplateDescription(availableSmsTemplatesList?.find(
+            (template: SmsTemplateType) => template.id === templateId)?.description);
+    };
+
+    const handleTemplateChange = (updatedTemplateAttributes: Partial<SmsTemplate>) => {
+        setSelectedSmsTemplate({ ...selectedSmsTemplate, ...updatedTemplateAttributes });
+        setIsTemplateNotAvailable(false);
+    };
+
+    const handleLocaleChange = (locale: string) => {
+        setCurrentSmsTemplate({ ...selectedSmsTemplate });
+        setIsTemplateNotAvailable(false);
+        setIsSystemTemplate(false);
+        setSelectedLocale(locale);
+    };
+
+    const handleSubmit = () => {
+
+        const template: SmsTemplate = {
+            ...selectedSmsTemplate,
+            id: selectedLocale.replace("-", "_")
+        };
+
+        if (isTemplateNotAvailable || isSystemTemplate) {
+            createNewSmsTemplate(selectedSmsTemplateId, template)
+                .then((_response: SmsTemplate) => {
+                    dispatch(addAlert<AlertInterface>({
+                        description: t("extensions:develop.smsTemplates.notifications.updateSmsTemplate" +
+                            ".success.description"),
+                        level: AlertLevels.SUCCESS,
+                        message: t("extensions:develop.smsTemplates.notifications.updateSmsTemplate" +
+                            ".success.message")
+                    }));
+                }).catch((error: IdentityAppsApiException) => {
+                    dispatch(addAlert<AlertInterface>({
+                        description: t("extensions:develop.smsTemplates.notifications.updateSmsTemplate" +
+                            ".error.description"),
+                        level: AlertLevels.ERROR,
+                        message: error.message ?? t("extensions:develop.smsTemplates.notifications" +
+                            ".updateSmsTemplate.error.message")
+                    }));
+                }).finally(() => smsTemplateMutate());
```

Comment:
Formatting suggestion. Check other places as well.
```suggestion
                })
                .catch((error: IdentityAppsApiException) => {
                    dispatch(addAlert<AlertInterface>({
                        description: t("extensions:develop.smsTemplates.notifications.updateSmsTemplate" +
                            ".error.description"),
                        level: AlertLevels.ERROR,
                        message: error.message ?? t("extensions:develop.smsTemplates.notifications" +
                            ".updateSmsTemplate.error.message")
                    }));
                })
                .finally(() => smsTemplateMutate());
```